### PR TITLE
Core/Scripts/DB: move Wyrmrest Temple taxi FP to SAI

### LINKED ISSUE #15957: Gossip texts for Wyrmrest Temple taxis are incorrect.
c703e01

The texts for the taxi flights between the levels of Wyrmrest Temple are incorrect as is. I tried to look at doing it properly versus just changing strings per Aokromes's suggestion in #15883, but I had to give up - just don't know enough about C++. Thus, decided to open an issue regarding it so that it can be tracked.

The texts are correct in the database already.

[Lord Afrasastrasz](http://www.wowhead.com/npc=27575/lord-afrasastrasz)
[Tariolstrasz](http://www.wowhead.com/npc=26443/tariolstrasz)
[Torastrasza](http://www.wowhead.com/npc=26949/torastrasza)

### PR DESCRIPTION
**Pull request by ghost**
Closes #15957

Move NPC gossip menu options from core script to DB SAI script
for the 3 local Flight Path NPCs in Wyrmrest Temple:
- 26443, Tariolstrasz (Steward of Wyrmrest Temple)
- 26949, Torastrasza (Majordomo to the Ruling Council)
- 27575, Lord Afrasastrasz (Commander of Wyrmrest Temple Defenses)

Additional positive effect: the NPCs are now using verified Blizzlike text in their gossip.

Extra credits:  @velinath for relevant background information in the issue,
@Helias & @ShinDarth for their excellent Keira2 web-based database editor.

### CODE REVIEW

**Comment by tkrokli** [comment]
OK, will do. I thought perhaps Wyrmrest Temple was changed at the end of Cataclysm, but didn't check.

**Comment by tkrokli** [comment]
This PR has been tested with a default character leveled up to 77. Gossip & actions work as expected.
The only problem/issue I can' t do anything about in this PR is the fact that the defined FP taxi zone/area for Lord Afrasastrasz is from the center of the NPC and behind him, not in front of him (issue #15398). The other 2 NPCs have got normal player taxi zone/area in front of them.

File: sql/updates/world/2015_12_09_12_world.sql [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(@Tariolstrasz, 0, 0, 1, 62, 0, 100, 0, 9455, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Tariolstrasz - On Gossip Option 0 Selected - Close Gossip'),
+(@Tariolstrasz, 0, 1, 0, 61, 0, 100, 0, 0, 0, 0, 0, 52, 878, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Tariolstrasz - On Gossip Option 0 Selected - Activate Taxi Path 878 Ground - Top'),
+(@Tariolstrasz, 0, 2, 3, 62, 0, 100, 0, 9455, 1, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Tariolstrasz - On Gossip Option 1 Selected - Close Gossip'),
+(@Tariolstrasz, 0, 3, 0, 61, 0, 100, 0, 0, 0, 0, 0, 52, 883, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Tariolstrasz - On Gossip Option 1 Selected - Activate Taxi Path 883 Ground - Middle');
```
Comment (joschiwald) [inline sql/updates/world/2015_12_09_12_world.sql]: one possible solution to have only 3 sai entries per npc: on gossip select -> activate taxi -> link to close gossip

Comment (tkrokli) [inline sql/updates/world/2015_12_09_12_world.sql]: Aha, right. Thanks, good point. I will make those changes soon and update the PR.

**Comment by tkrokli** [comment]
Worth noticing when using this script at the Wyrmrest Temple: if you get the DBError
```
CreatureTextMgr: Could not find Text for Creature Wyrmrest Temple Drake (Entry 26925, GUID 240) in 'creature_text' table. Ignoring.
```
... it is very likely that one or more (maybe all) of the gossip options stop working on the 3 NPCs.

The error message was reported in https://github.com/TrinityCore/TrinityCore/issues/15120#issue-95907943 as well as a closed issue.

**Comment by velinath** [comment]
@tkrokli I was looking today at migrating more of these NPCs in a similar fashion (ref velinath/TrinityCore@75441be and #16112). Maybe we should create a more general PR to do all possible cases for this rather than on a case by case basis. Do you think it is worthwhile? Would be happy to work with you on this or integrate your work into my branch to PR the whole thing when done.

**Comment by Aokromes** [comment]
One outdated list of things that needs to be converted: http://pastebin.com/x7wZLpxH
Anyway smaller PRs faster merges :)

**Comment by velinath** [comment]
works for me Aok, thanks for the input and a good todo list for me - will PR mine separately as soon as I have an opportunity to test

**Comment by tkrokli** [comment]
@velinath : Like Aok wrote, I think there is a better chance of getting a small case-by-case PR merged faster than grouping them all in 1 big PR, because it will both be easier to keep an overview of the changes in the case-by-case PRs as well as testing them separetely to see their effects in-game.

**Comment by Treeston** [comment]
Afrasastrasz seems bugged. It randomly worked once after a few tries, and hasn't worked since. Related #15398?

**Comment by Treeston** [comment]
Your SAI events have `SMART_EVENT_FLAG_NOT_REPEATABLE` set. This means the FPs will only work the first time someone selects them, then won't work until server reset.

**Comment by Treeston** [comment]
(And that was the wrong button.)

**Comment by tkrokli** [comment]
OK, thanks for pointing that out. I thought it was meant to stop the NPC from trying to repeat the same spellcast multiple times in a row, even after sending the player to the next FP. Will fix that ASAP.

**Comment by tkrokli** [comment]
Thanks for the useful feedback and assistance. :smiley:

**Pull request merged**